### PR TITLE
[#172172983] update ui when a new payment method is added

### PR DESCRIPTION
**Pull request by valentinacala**
**Short description:**
This pr adds new content displayed on WalletHomeScreen when the user adds a new payment method instead of the transaction list.

**Changes:**
- The WalletHomeScreen has a new navigation parameter to display/hide the component to be confirmation of a new saved payment method
- The withConditionedContent HOC is updated so to grant the navigation parameters of the wrapped component are inherited

FINAL RESULT:
addCard

HERE JUST THE NEW CONTENT:
image

### CODE REVIEW

**Comment by Undermaken** [comment]
@valentinacala could you solve branch conflicts?

File: ts/components/helpers/withConditionalView.tsx [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
   ConditionComponent: React.ComponentType<C>
 ) {
-  class ConditionalView extends React.PureComponent {
+  /**
+   * The component should inherit the navigation params of both the wrapped and
+   * the contition component to ensure the nagivigation can address the proper
+   * navigation paramaters
+   */
+  type nullN = {};
+  type NP = P extends NavigationInjectedProps<infer N> ? N : nullN;
```
Comment (Undermaken) [inline ts/components/helpers/withConditionalView.tsx:20]:
```suggestion
  type NP = P extends NavigationInjectedProps<infer N> ? N : never;
```

File: ts/components/helpers/withConditionalView.tsx [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
+  type NC = C extends NavigationInjectedProps<infer N> ? N : nullN;
```
Comment (Undermaken) [inline ts/components/helpers/withConditionalView.tsx:21]:
```suggestion
  type NC = C extends NavigationInjectedProps<infer N> ? N : never;
```

Comment (valentinacala) [reply ts/components/helpers/withConditionalView.tsx]: it works! updated within https://github.com/pagopa/io-app/pull/1668/commits/8494b7785a0e25e51e0791ee94f868920dfa1c24

File: ts/components/helpers/withConditionalView.tsx (hunk 1 above)
Comment (valentinacala) [reply ts/components/helpers/withConditionalView.tsx]: updated within https://github.com/pagopa/io-app/pull/1668/commits/8494b7785a0e25e51e0791ee94f868920dfa1c24

File: ts/components/helpers/withConditionalView.tsx (hunk 2 above)
Comment (valentinacala) [reply ts/components/helpers/withConditionalView.tsx]: it works! updated within https://github.com/pagopa/io-app/pull/1668/commits/8494b7785a0e25e51e0791ee94f868920dfa1c24

File: ts/components/helpers/withConditionalView.tsx (hunk 1 above)
Comment (valentinacala) [reply ts/components/helpers/withConditionalView.tsx]: updated within https://github.com/pagopa/io-app/pull/1668/commits/8494b7785a0e25e51e0791ee94f868920dfa1c24

Comment (valentinacala) [reply ts/components/helpers/withConditionalView.tsx:20]: ok, I found the critical point. When using never, the navigation params are not properly read from the screen wrapped with the HOC.

Here the error (on compile):
image

update reverted within https://github.com/pagopa/io-app/pull/1668/commits/08f3ebee9d47a38d2ef77644b559fddf31b26c66

File: ts/components/helpers/withConditionalView.tsx (hunk 2 above)
Comment (valentinacala) [reply ts/components/helpers/withConditionalView.tsx:21]: same above

**Pull request merged**